functions/utils/cache.ts

```diff
@@ -0,0 +1,25 @@
+class Cache {
+  async match(request: string) {
+    const cache = await caches.open('function-cache')
+    const response = await cache.match(request)
+    if (!response) return undefined
+    return JSON.parse(await response.text())
```

Comment:
Can you validate that `JSON.parse(await response.text())` matches the `Data` type, and throw if it does not?